app/models/alchemy/element.rb

```diff
@@ -321,9 +321,11 @@ def copy_nested_elements_to(target_element)
     #   Pass an element name to get previous of this kind.
     #
     def previous_or_next(dir, name = nil)
-      elements = page.elements.published.where("#{self.class.table_name}.position #{dir} #{position}")
+      dir = dir == '>' ? '>' : '<'
+      order = dir == '>' ? :asc : :desc
```

Comment:
Here's a proposed refactor (a similar pattern could also be used for `Alchemy::Page#next_or_previous`):

```ruby
...

private

def select_element(elements, name = nil, order)
  elements = elements.named(name) if name.present?
  elements.reorder(position: order).limit(1).first
end

def next_element(name = nil)
  elements = page.elements.published.where('position > ?', position)
  select_element(elements, name, :asc)
end

def previous_element(name = nil)
  elements = page.elements.published.where('position < ?', position)
  select_element(elements, name, :desc)
end
```

I've implemented this locally and all tests are passing. Do you think that this is an okay approach?